ddtrace/internal/datadog/profiling/dd_wrapper/include/profile.hpp

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include "types.hpp"
+
+#include <atomic>
+#include <deque>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <string_view>
+#include <unordered_set>
+#include <vector>
+
+extern "C"
+{
+#include "datadog/profiling.h"
+}
+
+namespace Datadog {
+
+constexpr unsigned int g_default_max_nframes = 128;
+
+// Unordered containers don't get heterogeneous lookup until gcc-10, so for now use this
+// strategy to dedup + store strings.
+using StringTable = std::unordered_set<std::string_view>;
+
+// Serves to collect individual samples, as well as lengthen the scope of string data
+class Profile
+{
+  private:
+    // Serialization for static state
+    // - string table
+    // - ddog_profile
+    std::atomic<bool> first_time{ true };
+    std::mutex profile_mtx{};
+    std::atomic<pid_t> current_pid{ 0 };
+
+    // Storage for strings
+    std::deque<std::string> string_storage{};
```

Comment:
I think we have already talked about this (sorry if I forgot the conclusion)
Why do we not make the storage the std::set (and put standard strings in there, as well as make string views hashable) ?

Reply:
Sorry--does this conversation answer your question?  https://github.com/DataDog/dd-trace-py/pull/6731/files/be1aac6146cf8cd614e49f93ba960e16d2d3710f#r1304402419

The basic premise is that the version of C++ we have in CI right now doesn't support heterogeneous access for sets, so the incoming `string_view` would need to get converted to `string` for comparison.  I think there are some fun and exciting ways around this, but I'm even more interested in removing the temporary storage entirely if we can help it.

Reply:
Well at least I knew I was going to be confused again.